lib/mojito.js

```diff
@@ -57,17 +57,23 @@ Mojito = (function () {
 
             var recipes = this.options.recipes,
                 totalSimpleRate,
-                recipe;
+                recipe,
+                oneRecipeRateSet,
+                oneRecipeRateNotSet;
 
             for (p in recipes) {
                 recipe = recipes[p];
                 recipe.id = p;
 
                 if (recipe.sampleRate != null) {
                     totalSimpleRate += recipe.sampleRate;
+                    oneRecipeRateSet = true;
+                } else {
+                    oneRecipeRateNotSet = true;
                 }
             }
 
+            if (oneRecipeRateSet && oneRecipeRateNotSet) throw new Error('The sampleRate property is set to one recipe object, but not others');
```

Comment:
Thinking of a more descriptive error message with suggested fix, e.g.
"Recipe-level sampleRate property must be defined for all recipes if it is being used"

Thoughts @kingo55?

Reply:
Yes, I think so.

Reply:
Cool, @allmywant - please change the error string to: "Recipe-level sampleRate property must be defined for all recipes if it is being used" and we should be right to approve + merge.

Reply:
@dapperdrop 
Error message updated.